Sources/_NIOConcurrency/AsyncAwaitSupport.swift

```diff
@@ -0,0 +1,189 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+extension NIO.System {
+    public static var hasAsyncAwaitSupport: Bool {
+        #if compiler(>=5.4) // we cannot write this on one line with `&&` because Swift 5.0 doesn't like it...
+        #if compiler(>=5.4) && $AsyncAwait
+        return true
+        #else
+        return false
+        #endif
+        #else
+        return false
+        #endif
+    }
+}
+
+#if compiler(>=5.4) // we cannot write this on one line with `&&` because Swift 5.0 doesn't like it...
+#if compiler(>=5.4) && $AsyncAwait
+import NIO
```

Comment:
```suggestion
import NIO
import _Concurrency
```

Reply:
It won't compile without it. However, if you're aiming for a release Swift version, which won't have this underscored module, it's fine :)

Reply:
Right, it only compiled if you passed `-Xfrontend -enable-experimental-concurrency` which I did. Now fixed. thanks